client/src/components/CreateNewTripPage.jsx

```diff
@@ -1,35 +1,117 @@
+                    thingstodoId,
+                    value.day,
+                    value.time
+                );
+            }
+        } catch (err) {
+            console.error(err);
+        }
+    };
 
     useEffect(() => {
+        const fetchExistingData = async () => {
+            const existingActivities = await fetchActivitesByTripId(tripId);
+            const existingActivityMap = existingActivities.reduce(
+                (acc, activity) => {
+                    acc[activity.thingstodoId] = {
+                        day: activity.tripDay,
+                        time: activity.timeOfDay,
+                    };
+                    return acc;
+                },
+                {}
+            );
+            setActivityDays(existingActivityMap);
+        };
+
+        fetchExistingData();
         fetchThingsToDo(setThingsToDo, parkId);
         fetchTripDetailsById(setTripData, tripId);
-    }, [fetchThingsToDo]);
+    }, [tripId, parkId]);
 
-    return(
+    return (
         <>
-        <h1>New Trip {tripId}: {tripData.name}</h1>
-        <p>Days: {tripData.days}</p>
-        <h2>Things to do:</h2>
-        {thingsToDo &&
-            thingsToDo.map(activity => (
-                <div key={activity.id}>
-                    <label htmlFor={activity.id}>{activity.name}</label>
-                    <p>{activity.description}</p>
-                    <select>
-                        {}
-                    </select>
-                </div>
-            )
-        )}
+            <h1>
+                New Trip {tripId}: {tripData.name}
+            </h1>
+            <p>Days: {tripData.days}</p>
+            <h2>Things to do:</h2>
+            <button onClick={handleSave}>Save</button>
+            {thingsToDo &&
+                thingsToDo.map((activity) => (
+                    <div key={activity.id} className="activity">
+                        <div>
+                            <h3>{activity.name}</h3>
+                            <p>{activity.description}</p>
+                        </div>
+                        <select
+                            value={activityDays[activity.id]?.day || ''}
+                            onChange={(e) =>
+                                setActivityDays({
+                                    ...activityDays,
+                                    [activity.id]: {
+                                        ...activityDays[activity.id],
+                                        day: parseInt(e.target.value),
+                                    },
+                                })
+                            }
+                        >
+                            <option value="">Select a day</option>
+                            {Array.from({ length: tripData.days }, (_, i) => (
+                                <option key={i} value={i + 1}>
+                                    Day {i + 1}
+                                </option>
+                            ))}
```

Comment:
Might be more manageable to do this:
```suggestion
                            {Array.from({ length: tripData.days }, (_, i) => 
                            {
                                const dayNum = i + 1;
                              return (
                                <option key={i} value={dayNum}>
                                    Day {dayNum}
                                </option>
                            );
                            })}
```